packages/notifier/src/notifier.js

```diff
@@ -0,0 +1,54 @@
+import { producePromise } from '@agoric/produce-promise';
+import harden from '@agoric/harden';
+
+/**
+ * Makes a notifier object, which allows a service to produce a stream of update
+ * promises. Each update is a record containing
+ *      { currentState, nextStatePromise, updateHandle }.
+ *   currentState is whatever state the service wants to publish,
+ *   nextStatePromise is a HandledPromise, which resolves when the state changes
```

Comment:
I am going to be insufferable pedantic about JS promise terminology because it's important to be precise. A promise is either
   * fulfilled
   * rejected
   * unresolved
   * forwarded to another promise.
"forwarded" is Midori terminology. JS doesn't have an explicit term for this. Once it is forwarded, then it makes a transition to fulfilled or rejected according to the promise it is forwarded to.

The groupings are

   * settled
     * fulfilled
     * rejected
  * unsettled
     * forwarded
     * unresolved

and

   * resolved
   * resolved
     * settled
       * fulfilled
       * rejected
     * forwarded
   * unresolved

For someone holding the promise, the transition from unresolved to resolved is not observable. The transition from unsettled to settled is.

The transition from resolved to unresolved is only useful from the pov of someone holding the resolver.

So

...is a HandledPromise, which settles (fulfills?) when the state changes.